server/src/main/java/org/opensearch/rest/MethodHandlers.java

```diff
@@ -43,7 +43,7 @@
  *
  * @opensearch.api
  */
-final class MethodHandlers {
+public final class MethodHandlers {
```

Comment:
Could we introduce the "view" only project for `MethodHandlers`? In this case we:
- could keep the `MethodHandlers` package private
- could have public interface with only readonly getters

Reply:
I extracted `MethodHandlers` as public with the two r/o getters. Take a look?

Not sure if there's a better way to cast an iterator of `RestMethodHandlers` onto one of `MethodHandlers` than this.

```java
    public Iterator<MethodHandlers> getAllHandlers() {
        List<MethodHandlers> methodHandlers = new ArrayList<>();
        handlers.retrieveAll().forEachRemaining(methodHandlers::add);
        return methodHandlers.iterator();
    }
```